tests/endtoend/test_eventgrid_functions.py

```diff
@@ -69,17 +86,82 @@ def test_eventgrid_trigger(self):
                 response = r.json()
 
                 self.assertEqual(
-                    response,
-                    {
-                        'id': data[0]['id'],
-                        'data': data[0]['data'],
-                        'topic': data[0]['topic'],
-                        'subject': data[0]['subject'],
-                        'event_type': data[0]['eventType'],
-                    }
+                    response, {'id': data[0]['id'], 'data': data[0]['data'],
+                               'topic': data[0]['topic'],
+                               'subject': data[0]['subject'],
+                               'event_type': data[0]['eventType']}
                 )
             except AssertionError:
                 if try_no == max_retries - 1:
                     raise
             else:
                 break
+
+    @testutils.retryable_test(1, 5)
+    def test_eventgrid_output_binding(self):
+        """test event_grid output binding
+
+        This test needs three functions to work.
+        1. `eventgrid_output_binding`
+        2. `eventgrid_output_binding_message_to_blobstore`
+        3. `eventgrid_output_binding_success`
+
+        This test calls the eventgrid_output_binding function, sends in a unique
+        uuid as `data` in the body to the httpTrigger which sends in that value
+        in the eventGrid output data. The eventGrid topic is configured to
+        send the event to a storage queue.
+
+        The second function (`eventgrid_output_binding_message_to_blobstore`)
+        reads from that storage queue and puts into a blob store.
+
+        The third function (`eventgrid_output_binding_success`) reads the
+        text from the blob store and compares with the expected result. The
+        unique uuid should confirm if the message went through correctly to
+        EventGrid and came back as a blob.
+        """
+
+        test_uuid = uuid.uuid4().__str__()
+
+        data = "{" + "'test_uuid': '{0}'".format(test_uuid) + "}"
+        expected_response = "Sent event with subject: {}, id: {}, data: {}, " \
+                            "event_type: {} to EventGrid!".format(
+                                "test-subject", "test-id", data, "test-event-1")
+        expected_final_data = {
+            'id': 'test-id', 'subject': 'test-subject', 'dataVersion': '1.0',
+            'eventType': 'test-event-1',
+            'data': {'test_uuid': test_uuid}
+        }
```

Comment:
I think this is pretty cool to place expected result at the top of a test case, I will follow this flow when writing test cases.

Reply:
It depends :)
In some cases having expected value closer to the assert makes sense.